homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,485 @@
+        key="power_consumption_battery",
+        translation_key="power_consumption_battery",
+        icon="mdi:home-battery",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="energy_consumed_grid",
+        translation_key="energy_consumed_grid",
+        icon="mdi:transmission-tower",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_consumption_grid",
+        translation_key="power_consumption_grid",
+        icon="mdi:transmission-tower",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_registry_enabled_default=False,
+    ),
+]
+SENSOR_DESCRIPTIONS = {
+    "SOLAR_PV": [
+        SensorEntityDescription(
+            key="energy_supplied",
+            translation_key="solar_pv_energy_supplied",
+            icon="mdi:solar-power",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+        ),
+        SensorEntityDescription(
+            key="power_supply",
+            translation_key="solar_pv_power_supply",
+            icon="mdi:solar-power",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+            device_class=SensorDeviceClass.POWER,
+        ),
+    ],
+    "WATER_HEATER": (
+        SENSOR_DESCRIPTIONS_GENERIC_CONSUMER
+        + [
+            SensorEntityDescription(
+                key="available_energy",
+                translation_key="water_heater_available_energy",
+                state_class=SensorStateClass.MEASUREMENT,
+                native_unit_of_measurement=UnitOfVolume.LITERS,
+                device_class=SensorDeviceClass.WATER,
+            ),
+            SensorEntityDescription(
+                key="max_energy",
+                translation_key="water_heater_max_energy",
+                state_class=SensorStateClass.MEASUREMENT,
+                native_unit_of_measurement=UnitOfVolume.LITERS,
+                device_class=SensorDeviceClass.WATER,
+            ),
+            SensorEntityDescription(
+                key="output_temp",
+                translation_key="water_heater_output_temp",
+                state_class=SensorStateClass.MEASUREMENT,
+                native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+                device_class=SensorDeviceClass.TEMPERATURE,
+            ),
+            SensorEntityDescription(
+                key="temp_sensor1",
```

Comment:
Please explain the value of the sensors 1 to 6? Why can't we name them better?

Reply:
This is the named used in the combined energy API. I can't change it.

Reply:
What temperature are these sensors describing?

For the user it would be better in my opinion if the sensor has a descriptive name

Reply:
There are up to 6 sensors with 1 at the bottom to n at the top. 1-6 actually makes sense.